app/models/partnership_csv_upload.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class PartnershipCsvUpload < ApplicationRecord
+  has_paper_trail
+  has_one_attached :csv
+  belongs_to :lead_provider, optional: true
+  belongs_to :delivery_partner, optional: true
+
+  validate :csv_validation
+
+  MAX_FILE_SIZE = 2.megabytes
+
+private
+
+  def csv_validation
+    return unless csv.attached?
+
+    if csv.filename.extension != "csv"
```

Comment:
Does this get folded to lowercase. Wouldn't it be better to use "filemagic" to determine the base file structure rather than rely on an extension. You could always (ugh) shell out if there isn't an acceptable gem.

Reply:
Will this definitely work, as I think the file becomes an ActiveStorage::object once it is attached ?

Reply:
Would that definitely work, as the file becomes an ActiveStorage::object once it is attached?